src/ControlSystem/Actions/Initialization.hpp

```diff
@@ -0,0 +1,72 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <string>
+#include <tuple>
+#include <utility>
+
+#include "ControlSystem/Averager.hpp"
+#include "ControlSystem/Controller.hpp"
+#include "ControlSystem/Tags.hpp"
+#include "ControlSystem/TimescaleTuner.hpp"
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace Initialization {
+namespace Actions {
+/*!
+ * \ingroup ControlSystemGroup
+ * \ingroup InitializationGroup
+ * \brief Initialize items related to the control system
+ *
+ * DataBox:
+ * - Uses: Nothing
+ * - Adds: Nothing
+ * - Removes: Nothing
+ * - Modifies:
+ *   - `control_system::Tags::ControlSystemName`
+ *
+ * \note This action relies on the `SetupDataBox` aggregated initialization
+ * mechanism, so `Actions::SetupDataBox` must be present in the `Initialization`
+ * phase action list prior to this action.
+ */
+template <typename Metavariables, typename ControlSystem>
+struct InitializeControlSystem {
+  // The averager here and controller below are hard coded with a DerivOrder=2
+  // at the moment, because this is the DerivOrder of most functions of time in
+  // the domain creators. If we want to choose the DerivOrder at runtime (i.e.
+  // in an input file), that capability will need to be added later.
+  using initialization_tags = tmpl::list<control_system::Tags::Averager<2>,
+                                         control_system::Tags::TimescaleTuner>;
+
+  using initialization_tags_to_keep = initialization_tags;
+
+  using simple_tags = tmpl::flatten<
+      tmpl::append<tmpl::list<typename ControlSystem::simple_tags>,
+                   control_system::Tags::Controller<2>>>;
```

Comment:
I am surprised this compiles.  I think you want `using simple_tags = tmpl::push_back<typename ControlSystem::simple_tags, control_system::Tags::Controller<2>>;`.

Reply:
`tmpl::append` takes lists as arguments.  `control_system::Tags::Controller<2>` is not a list.  (Also, `simple_tags` is already a list, but that should still have worked.)

Reply:
Following up on this, have you actually used this code locally?  Testing this sort of thing on CI is hard until the whole system is in, but it should be tested somewhere before it is merged.

Reply:
I tested a slightly different version on a different branch (basically I made the Controller actually have options so I added it in with the initialization tags, but since it doesn't have any real options yet I just put it with the simple tags here), but no I forgot to test that here. I'll add an initialization test that just checks things have been added to the box properly.